instance.tf

```diff
@@ -1,8 +1,8 @@
 resource "google_sql_database_instance" "instance" {
   database_version     = var.database_version
-  name                 = var.instance_name
+  name                 = "${var.project_prefix}-${var.instance_name}"
   deletion_protection  = var.deletion_protection
-  master_instance_name = var.primary_instance_name
+  master_instance_name = "${var.primary_instance_name}" != null ? "${var.project_prefix}-${var.primary_instance_name}" : null
```

Comment:
If `project_prefix` is `null`, you don't want the `-` separator. As that would make the name start with a dash, which is both undesired and likely not supported either.

The cleanest solution might be to move the conditional logic to some locals and then refer to those locals here instead. If that'd be desired kinda depends on the amount of "magic" required to get the proper values.

Reply:
Current "fix" is not what I meant. What I meant was:

If `project_prefix` is defined, set `name` to `instance_name` prefixed with `project_prefix`, separated with a dash.
If `project_prefix` is not defined, just set `name` to `instance_name` (and thus no dash).

Similar logic for `master_instance_name`, which will then need nested conditional logic (as the value depends on both the presence of `primary_instance_name` and `project_prefix`).